src/CommunityPatch/Patches/Perks/Endurance/Riding/NomadicTraditionsPatch.cs

```diff
@@ -77,17 +77,20 @@ public sealed class NomadicTraditionsPatch : PatchBase<NomadicTraditionsPatch> {
         }
         float mountedFootman = Math.Min(availableHorses, troopsOnFoot);
         var mountedFootmanRatio = totalTroops == 0 ? 0 : mountedFootman / totalTroops;
-        var perkMountedFootmanRatio = GetPerkBaseRatio(0.3f) * mountedFootmanRatio;
+        var mountedFootmanMagicNumber = (float)MountedFootmenRatioModifierMethodInfo.Invoke(__instance, new object[] {1, 1});
+        var perkMountedFootmanRatio = GetPerkBaseRatio(mountedFootmanMagicNumber) * mountedFootmanRatio;
         
         var cavalryRatio = totalTroops == 0 ? 0 : mountedTroops / totalTroops;
-        var perkCavalryRatio = GetPerkBaseRatio(0.6f) * cavalryRatio;
-        
-        var baseSpeedMethod = typeof(DefaultPartySpeedCalculatingModel).GetMethod("CalculateBaseSpeedForParty", BindingFlags.Instance | BindingFlags.NonPublic);
-        var baseSpeed = (float)baseSpeedMethod.Invoke(__instance, new object[] { totalTroops });
+        var cavalryMagicNumber = (float)CavalryRatioModifierMethodInfo.Invoke(__instance, new object[] {1, 1});
+        var perkCavalryRatio = GetPerkBaseRatio(cavalryMagicNumber) * cavalryRatio;
 
-        var perkBonus = (perkCavalryRatio + perkMountedFootmanRatio) * baseSpeed;
+        var baseSpeed = (float)BaseSpeedMethodInfo.Invoke(__instance, new object[] {totalTroops});
+        
+        var explainedNumber = new ExplainedNumber(baseSpeed, explanation);
+        explainedNumber.AddFactor(perkCavalryRatio, ActivePatch._perk.Name);
+        explainedNumber.AddFactor(perkMountedFootmanRatio, ActivePatch._perk.Name);
+        var perkBonus = explainedNumber.ResultNumber - baseSpeed;
         __result += perkBonus;
```

Comment:
I'm not very confident is this. It does what I want and it gives the correct result, but is it the right way to go about it?

Reply:
Can you take a screenshot of the explainer tooltip that this applies to?

I think since BaseSpeed is a const (CIL nomenclature; "static literal valuetype field") on the class, you might be able to declare a parameter that Harmony can extract for you; as `float ___BaseSpeed` instead of having to grab it by reflection.

In TW's implementation of `CalculatePureSpeed` you can see a pattern for this;
```cs
      ExplainedNumber bonuses = new ExplainedNumber(this.CalculateBaseSpeedForParty(num), explanation, (TextObject) null);
      float cavalryRatioModifier = this.GetCavalryRatioModifier(num, totalCavalryCount1);
      bonuses.AddFactor(cavalryRatioModifier, DefaultPartySpeedCalculatingModel._textCavalry);
...
return bonuses.ResultNumber;
```

`ExplainedNumber` may add a "Base" component to the explainer tooltip that you may need to remove; other patches do that, since you're using it to modify based on the base speed.
If you want to apply the factors manually, you can use Add and do the factor multiplication step separately against the `BaseSpeed`, so the first parameter to your `ExplainedNumber` can be `__result`, then you just `return __result`.

Reply:
![Screenshot_37](https://user-images.githubusercontent.com/5471818/79998657-59352180-8491-11ea-92bf-9d7b32344cff.png)